htlcswitch/link.go

```diff
@@ -1345,7 +1346,7 @@ func (l *channelLink) htlcManager() {
 		// With our link's in-memory state fully reconstructed, spawn a
 		// goroutine to manage the reclamation of disk space occupied by
 		// completed forwarding packages.
-		l.Wg.Add(1)
+		l.cg.WgAdd(1)
```

Comment:
`channelLink` starts a fixed number of goroutines (1 or 2) in its `Start` method, so a regular `sync.WaitGroup` can be used here. While using `ContextGuard`'s internal wait group works, it appears to be repurposed for something it wasn't designed for. I suggest adding a regular `WaitGroup` alongside `ContextGuard` and using it instead of `WgAdd` here and when starting the `htlcManager` goroutine.

Reply:
i think let's just keep this PR a refactor though?